x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/Mapper.java

```diff
@@ -161,13 +162,9 @@ private PhysicalPlan map(UnaryPlan p, PhysicalPlan child) {
         }
 
         if (p instanceof Lookup lookup) {
-            return new HashJoinExec(
-                lookup.source(),
-                child,
-                lookup.tableName(),
-                lookup.matchFields(),
-                lookup.matchValues(),
-                lookup.mergeValues()
+            return new HashJoinExec(lookup.source(), child, null, emptyList(), null, null
+            // lookup.matchValues(),
+            // lookup.mergeValues()
```

Comment:
This doesn't match anymore since there's no more Lookup - just join and local relationship.
This means the hashjoin can be broadcasted (as data strategy).